hyperjumptech/monika
# Fix: do not overwrite existing config

# Monika Pull Request (PR)  
fix #306 
## What feature/issue does this PR add  
When using --create-config with --har the existing monika.json file that was lovingly crafted is overwritten without any warning.
Overwritting files without any warning or confirmation is not a good user experience.

## How did you implement / how did you fix it  
1.  do not continue writing config file if it is already exists
2. overwrite config file if force flag is given

## How to test  
1. run monika using flags `./bin/run --create-config --har ./src/components/config/__tests__/form_encoded.har --output monika.config.json`, it should show error message if output file is already exist. e.g run 2 times using the same command
2. run monika using flags `./bin/run --create-config --har ./src/components/config/__tests__/form_encoded.har --output monika.config.json --force`, it should overwrite the existing config file. Should show message info `INFO: monika.config.json file has been created.`
